scripts/clean-shrinkwrap.js

```diff
@@ -78,6 +78,13 @@ traverseDeps(shrinkwrap, (parent, parentName, name, info) => {
     return
   }
 
+  // Ensure the `resolved` field contains a HTTPS registry URL,
+  // as some versions of npm apparently fall back to HTTP on some platforms
+  // under some circumstances
+  if (/^http:\/\//.test(info.resolved)) {
```

Comment:
I wonder if this should be restricted to checking for `http://registry.npmjs.org/` ? Is it possible that (potentially) some module might come from a different non-npmjs registry that only serves http and not https ?

FYI:
```
$ grep resolved npm-shrinkwrap.json | grep -v registry
      "resolved": "git://github.com/tessel/node-usb.git#38cc9cc75759e74f3d3ee8c79ca852395c3529b0",
```

Reply:
> from a different non-npmjs registry that only serves http and not https ?

not happening in our cases; you'd have to configure your local npm to use a different registry – and even then we'd want the shrinkwrap to point to what's been shrinkwrapped.

> FYI: grep resolved npm-shrinkwrap.json | grep -v registry

This tests whether the resolved URL even starts with `http://`, so will leave those alone, which is what we want.

Reply:
> This tests whether the resolved URL even starts with http://, so will leave those alone, which is what we want.

Yeah, I was more pointing it out in a "we already have one module that doesn't come from `https://registry.npmjs.org/` (which in this case comes from a `git://` URL), I wonder if in future we might have other modules that come from a `http://` URL that doesn't have a `https://` counterpart?" style :wink: